cairo/tests/ethereum/cancun/vm/instructions/test_system.py

```diff
@@ -31,8 +33,45 @@ def system_strategy(draw):
     return evm
 
 
+@composite
+def beneficiary_from_state(draw):
+    """Generate test cases for selfdestruct instruction.
+
+    This strategy generates an EVM instance with a beneficiary address on the stack:
+    - 80% chance: address from state (existing accounts)
+    - 20% chance: random address
+    - Always pushes the address to the stack
+    """
+    evm = draw(
+        EvmBuilder()
+        .with_env()
+        .with_message()
+        .with_stack()
+        .with_gas_left()
+        .with_running()
+        .with_accessed_addresses()
+        .with_accessed_storage_keys()
+        .with_accounts_to_delete()
+        .with_touched_accounts()
+        .build()
+    )
+
+    # Choose between state address (80%) or random address (20%)
+    use_state_address = draw(st.integers(0, 99)) < 80
+
+    if use_state_address and evm.env.state._main_trie._data:
+        # Get address from state if av
+        beneficiary = draw(st.sampled_from(list(evm.env.state._main_trie._data.keys())))
+    else:
+        beneficiary = draw(st.from_type(Address))
```

Comment:
how often do you get the case beneficiary == originator?

Reply:
nevertheless its easy to add so I added it

Reply:
i agree, i'm simply asking if all cases are handled, what would greatly help is Codecov, so we can see at a glance which lines of code are covered